components/ConversationsList.tsx

```diff
@@ -112,6 +112,26 @@ const ConversationsList = ({
       getLatestMessage(convoBMessages)?.sent || new Date()
     return convoALastMessageDate < convoBLastMessageDate ? 1 : -1
   }
+
+  const [resetPage, setResetPage] = useState(false)
+
+  const reloadIfQueryParamPresent = () => {
+    if (window.location.pathname) {
+      const path = window.location.pathname.replace('/dm/', '')
+      const matchAddress = conversations.filter(
+        (convo) => path == convo.peerAddress
+      )
+      if (Array.isArray(matchAddress) && matchAddress.length > 0) {
+        setResetPage(!resetPage)
+        router.push(window.location.pathname)
+      }
```

Comment:
Yes, that makes sense! Ideally there would be some feedback or hint to the user that they were redirected to '/' because the address was invalid. I don't believe we have designs for that - we should check with @darickdang when he gets back - but either way I would not consider it a ship blocker.

Reply:
Have worked on this and tried to address the above conditions. Thank you @elisealix22 for highlighting the edge cases